hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/HederaStateModule.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.state;
+
+import dagger.Module;
+import dagger.Provides;
+import javax.inject.Singleton;
+
+@Module
+public interface HederaStateModule {
+    @Provides
+    @Singleton
+    static HederaState provideWorkingHederaState(WorkingStateAccessor workingStateAccessor) {
```

Comment:
With `@Singleton` scope, Dagger will only call this provision method _once_; which means we'll get only the first mutable state.

Instead we should should have a `WorkingStateAccessor provideWorkingStateAccessor()` and inject `WorkingStateAccessor` into the components that need access to the working state.

Reply:
Fixed. Thanks Michael !